ThePrimeagen/init.vim

```diff
@@ -166,3 +166,12 @@ augroup END
 
 autocmd BufWritePre * :call TrimWhitespace()
 
+function! CleanExtraSpaces() "Function to clean unwanted spaces
```

Comment:
TrimWhitespace removes any extraneous whitespace at the end of the lines.  What exactly is this doing differently?  I am curious, perhaps TrimWhitespace isn't enough.